consensus/ibft/src/test/java/tech/pegasys/pantheon/consensus/ibft/IbftGossipTest.java

```diff
@@ -0,0 +1,271 @@
+
+import java.math.BigInteger;
+import java.util.Optional;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class IbftGossipTest {
+  private IbftGossip ibftGossip;
+  @Mock private IbftNetworkPeers ibftNetworkPeers;
+
+  @Before
+  public void setup() {
+    ibftGossip = new IbftGossip(ibftNetworkPeers);
+  }
+
+  private SignedData<ProposalPayload> createSignedProposalPayload(final KeyPair signerKeys) {
+    final MessageFactory messageFactory = new MessageFactory(signerKeys);
+    final ConsensusRoundIdentifier roundIdentifier =
+        new ConsensusRoundIdentifier(0x1234567890ABCDEFL, 0xFEDCBA98);
+    final Block block =
+        TestHelpers.createProposalBlock(singletonList(AddressHelpers.ofValue(1)), 0);
+    return messageFactory.createSignedProposalPayload(roundIdentifier, block);
+  }
+
+  private SignedData<PreparePayload> createSignedPreparePayload(final KeyPair signerKeys) {
+    final MessageFactory messageFactory = new MessageFactory(signerKeys);
+    final ConsensusRoundIdentifier roundIdentifier =
+        new ConsensusRoundIdentifier(0x1234567890ABCDEFL, 0xFEDCBA98);
+    return messageFactory.createSignedPreparePayload(
+        roundIdentifier, Hash.fromHexStringLenient("0"));
+  }
+
+  private SignedData<CommitPayload> createSignedCommitPayload(final KeyPair signerKeys) {
+    final MessageFactory messageFactory = new MessageFactory(signerKeys);
+    final ConsensusRoundIdentifier roundIdentifier =
+        new ConsensusRoundIdentifier(0x1234567890ABCDEFL, 0xFEDCBA98);
+    return messageFactory.createSignedCommitPayload(
+        roundIdentifier,
+        Hash.fromHexStringLenient("0"),
+        Signature.create(BigInteger.ONE, BigInteger.TEN, (byte) 0));
+  }
+
+  private SignedData<RoundChangePayload> createSignedRoundChangePayload(final KeyPair signerKeys) {
+    final MessageFactory messageFactory = new MessageFactory(signerKeys);
+    final ConsensusRoundIdentifier roundIdentifier =
+        new ConsensusRoundIdentifier(0x1234567890ABCDEFL, 0xFEDCBA98);
+    return messageFactory.createSignedRoundChangePayload(roundIdentifier, Optional.empty());
+  }
+
+  private SignedData<NewRoundPayload> createSignedNewRoundPayload(final KeyPair signerKeys) {
+    final MessageFactory messageFactory = new MessageFactory(signerKeys);
+    final ConsensusRoundIdentifier roundIdentifier =
+        new ConsensusRoundIdentifier(0x1234567890ABCDEFL, 0xFEDCBA98);
+    final SignedData<ProposalPayload> proposalPayload = createSignedProposalPayload(signerKeys);
+    return messageFactory.createSignedNewRoundPayload(
+        roundIdentifier, new RoundChangeCertificate(newArrayList()), proposalPayload);
+  }
+
+  private <P extends Payload> void rebroadcastToAllExceptSigner(
```

Comment:
since these methods are also asserting the rebroadcast it could be clearer for them to start with assert or something like that to make it it's an assertion